presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/BuiltInQueryPreparer.java

```diff
@@ -0,0 +1,167 @@
+            throws ParsingException, PrestoException, SemanticException
+    {
+        Statement wrappedStatement = sqlParser.createStatement(query, analyzerOptions.getParsingOptions());
+        if (warningCollector.hasWarnings() && analyzerOptions.getWarningHandlingLevel() == AS_ERROR) {
+            throw new PrestoException(WARNING_AS_ERROR, format("Warning handling level set to AS_ERROR. Warnings: %n %s",
+                    warningCollector.getWarnings().stream()
+                            .map(PrestoWarning::getMessage)
+                            .collect(joining(System.lineSeparator()))));
+        }
+        return prepareQuery(analyzerOptions, wrappedStatement, preparedStatements);
+    }
+
+    public BuiltInPreparedQuery prepareQuery(AnalyzerOptions analyzerOptions, Statement wrappedStatement, Map<String, String> preparedStatements)
+            throws ParsingException, PrestoException, SemanticException
+    {
+        Statement statement = wrappedStatement;
+        Optional<String> prepareSql = Optional.empty();
+        if (statement instanceof Execute) {
+            String preparedStatementName = ((Execute) statement).getName().getValue();
+            prepareSql = Optional.ofNullable(preparedStatements.get(preparedStatementName));
+            String query = prepareSql.orElseThrow(() -> new PrestoException(NOT_FOUND, "Prepared statement not found: " + preparedStatementName));
+            statement = sqlParser.createStatement(query, analyzerOptions.getParsingOptions());
+        }
+
+        if (statement instanceof Explain && ((Explain) statement).isAnalyze()) {
+            Statement innerStatement = ((Explain) statement).getStatement();
+            Optional<QueryType> innerQueryType = StatementUtils.getQueryType(innerStatement.getClass().getSimpleName());
+            if (!innerQueryType.isPresent() || innerQueryType.get() == QueryType.DATA_DEFINITION) {
+                throw new PrestoException(NOT_SUPPORTED, "EXPLAIN ANALYZE doesn't support statement type: " + innerStatement.getClass().getSimpleName());
+            }
+        }
+        List<Expression> parameters = ImmutableList.of();
+        if (wrappedStatement instanceof Execute) {
+            parameters = ((Execute) wrappedStatement).getParameters();
+        }
+        validateParameters(statement, parameters);
+        Optional<String> formattedQuery = Optional.empty();
+        if (analyzerOptions.isLogFormattedQueryEnabled()) {
+            formattedQuery = Optional.of(getFormattedQuery(statement, parameters));
+        }
+        return new BuiltInPreparedQuery(wrappedStatement, statement, parameters, formattedQuery, prepareSql);
+    }
+
+    private static String getFormattedQuery(Statement statement, List<Expression> parameters)
+    {
+        String formattedQuery = formatSql(
+                statement,
+                parameters.isEmpty() ? Optional.empty() : Optional.of(parameters));
```

Comment:
Remove the ternary. Optional.of(parameters) handles both cases, and you probably don't need Optional here either